modules/luci-mod-dashboard/htdocs/luci-static/resources/view/dashboard/include/20_lan.js

```diff
@@ -0,0 +1,150 @@
+'use strict';
+'require baseclass';
+'require rpc';
+'require network';
+
+var callLuciDHCPLeases = rpc.declare({
+	object: 'luci-rpc',
+	method: 'getDHCPLeases',
+	expect: { '': {} }
+});
+
+return baseclass.extend({
+	title: _('DHCP Devices'),
+
+	params: {},
+
+	load: function() {
+		return Promise.all([
+			callLuciDHCPLeases(),
+			network.getDevices()
+		]);
+	},
+
+	renderHtml: function() {
+
+		var container_wapper = E('div', { 'class': 'router-status-lan dashboard-bg box-s1' });
+		var container_box = E('div', { 'class': 'lan-info devices-list' });
+		var container_devices = E('div', { 'class': 'table assoclist devices-info' }, [
+			E('div', { 'class': 'tr table-titles  dashboard-bg' }, [
+				E('div', { 'class': 'th nowrap' }, _('Hostname')),
+				E('div', { 'class': 'th' }, _('IP Address')),
+				E('div', { 'class': 'th' }, _('MAC')),
+			])
+		]);
+
+		var container_deviceslist = E('div', { 'class': 'table assoclist devices-info' });
+
+		container_box.appendChild(E('div', { 'class': 'title'}, [
+			E('img', {
+				'src': L.resource('view/dashboard/icons/devices.svg'),
+				'width': 55,
+				'title': this.title,
+				'class': 'middle'
+			}),
+			E('h3', this.title)
+		]));
+
+		for(var idx in this.params.lan.devices) {
```

Comment:
Currently this list is neither sorted alphabetically on hostname or numerically on IP. I think for most users the primer is easiest to understand. 

Is the list currently ordered by age?

Reply:
Sorted by age, does it refer to the expires time of the device? 

I don't know if the following code can meet your suggestion.

`		leases.map(function(lease) {
			devices[lease.expires] = {
				hostname: lease.hostname || '?',
				ipv4: lease.ipaddr || '-',
				macaddr: lease.macaddr || '00:00:00:00:00:00',
			};
		});`

It will be sorted according to the devices lease expiration time from smallest to largest. That is, the order of the devices connected to the router.

Reply:
Sorted by age, does it refer to the expires time of the device? 

I don't know if the following code can meet your suggestion.

````
leases.map(function(lease) {
			devices[lease.expires] = {
				hostname: lease.hostname || '?',
				ipv4: lease.ipaddr || '-',
				macaddr: lease.macaddr || '00:00:00:00:00:00',
			};
		});
````

It will be sorted according to the devices lease expiration time from smallest to largest. That is, the order of the devices connected to the router.

Reply:
The current LuCI status page does not seem to have any particular ordering (from a very quick look)

I also suggest sorting by hostname so that it stays stable over time.

Reply:
It's sorted by lease and therefore newly added devices are on top. I think that good when people e.g. connect their smart TV and it appears I too of the list. I'd suggest to just keep it as is.

Reply:
So, should I change back to the previous version?

Reply:
Yes please.